src/artstationapplication/Polygon.java

```diff
@@ -51,6 +52,7 @@
         for(int i = 0; i < Integer.valueOf(input[8]); i++){
```

Comment:
Because polygons can have any number of vertices, there is no guarantee that index 14 will actually hold the name of the Polygon. I recommend moving the name to just before the vertex loop at index 8, and increasing the next two indices to 9 and 10. Then update the save method to match.